### PR TITLE
#1-4 Tasks. Dmitrii

### PR DESCRIPTION
**Pull request by Dmitrii-a11**
https://travis-ci.org/github/Dmitrii-a11/js-assignments

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ function getStringLength(value) {
  *   'Chuck','Norris'  => 'Hello, Chuck Norris!'
  */
 function getStringFromTemplate(firstName, lastName) {
-    throw new Error('Not implemented');
+    return new String(`Hello, ${firstName} ${lastName}!`);
```
Comment (FedorovD) [inline task/01-strings-tasks.js:58]: Should return `string` primitive type
String(...) will simply convert the expression into a boolean primitive value, while new String(...) will create a wrapper object around the converted string value

File: task/01-strings-tasks.js [hunk 2]
```diff
@@ -69,7 +69,9 @@ function getStringFromTemplate(firstName, lastName) {
  *   'Hello, Chuck Norris!' => 'Chuck Norris'
  */
 function extractNameFromTemplate(value) {
-    throw new Error('Not implemented');
+    let temp = value.replace(' ', '');
+    let ln = 'Hello,'.length;
+    return temp.substring(ln, temp.length - 1);
 }
```
Comment (FedorovD) [inline task/01-strings-tasks.js:75]: looks good, but please note substring can work with negative numbers also to take symbols from end. In the case `substring(7, -1);` can do the same.

File: task/01-strings-tasks.js [hunk 3]
```diff
@@ -114,7 +116,17 @@ function removeLeadingAndTrailingWhitespaces(value) {
  *   'cat', 3 => 'catcatcat'
  */
 function repeatString(value, count) {
-    throw new Error('Not implemented');
+    let retString = '';
+
+    if (eval(count) > 0) {
+        for (var i = 0; i < count; i++) {
```
Comment (FedorovD) [inline task/01-strings-tasks.js:122]: `var` is deprecated. It has no sense to use for browsers code even currently.

File: task/02-numbers-tasks.js [hunk 4]
```diff
@@ -89,7 +89,11 @@ function getDistanceBetweenPoints(x1, y1, x2, y2) {
  *   5*x = 0         => 0
  */
 function getLinearEquationRoot(a, b) {
-    throw new Error('Not implemented');
+    if (a != 0)
```
Comment (FedorovD) [inline task/02-numbers-tasks.js:92]: operator `===` or `!==` is always preferable

File: task/04-arrays-tasks.js [hunk 5]
```diff
@@ -200,7 +241,21 @@ function getTail(arr, n) {
...
+            ind++;
+        }
+        else
+            str += x;
+
+        return 0;
+    });
+
+    return str;
 }
```
Comment (FedorovD) [inline task/04-arrays-tasks.js:259]: `var` is deprecated. It has no sense to use for browsers code even currently.

**Comment by FedorovD** [comment]
Please, fix all `var` and `===` issues,
Also pay attention to javascript code style https://javascript.info/coding-style and https://learn.javascript.ru/variable-names
**I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!**

File: task/01-strings-tasks.js (hunk 2 above)
Comment (Dmitrii-a11) [reply task/01-strings-tasks.js:75]: Hello, FedorovD! From developer.mozilla: "Any argument value that is less than 0 ... is treated as if it were 0 ... .". So, for example: substring(-1,2) equals substring(0,2).  And in my case function with substring(7,-1) cannot pass the test.

**Comment by FedorovD** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**